modules/gdscript/gdscript_reconstructor.cpp

```diff
@@ -0,0 +1,885 @@
+				data.output_tab_required = true;
+				sz += "\n";
+				output_node(dic->elements[n].key, Output(out.indent + 1, &sz));
+				sz += draw_operator(" : ");
+				output_node(dic->elements[n].value, Output(out.indent + 1, &sz));
+				if (n && (n != dic->elements.size() - 1)) {
+					sz += draw_operator(", ");
+				}
+			}
+			sz += String("\n") + String("\t").repeat(out.indent) + draw_operator("}");
+		} break;
+		case GDScriptParser::Node::TYPE_ARRAY: {
+			GDScriptParser::ArrayNode *arr = (GDScriptParser::ArrayNode *)p_node;
+			sz += draw_operator("[");
+			for (int n = 0; n < arr->elements.size(); n++) {
+				output_node(arr->elements[n], Output(out.indent, &sz));
+				if (n != arr->elements.size() - 1) {
+					sz += draw_operator(", ");
+				}
+			}
+			sz += draw_operator("]");
+		} break;
+		case GDScriptParser::Node::TYPE_TYPE: {
+			const GDScriptParser::TypeNode *tp = (const GDScriptParser::TypeNode *)p_node;
+			sz += draw_operator(" : ") + draw_keyword(Variant::get_type_name(tp->vtype));
+		} break;
+		case GDScriptParser::Node::TYPE_BLOCK: {
+			return output_block((const GDScriptParser::BlockNode *)p_node, out);
+		} break;
+		case GDScriptParser::Node::TYPE_CLASS: {
+			return output_class((const GDScriptParser::ClassNode *)p_node, out);
+		} break;
+		case GDScriptParser::Node::TYPE_IDENTIFIER: {
+			const GDScriptParser::IdentifierNode *ident = (const GDScriptParser::IdentifierNode *)p_node;
+			sz += draw_identifier(ident->name);
+		} break;
+		case GDScriptParser::Node::TYPE_OPERATOR: {
+			return output_operator((const GDScriptParser::OperatorNode *)p_node, out, p_apply_parent_brackets);
+		} break;
+		case GDScriptParser::Node::TYPE_CONSTANT: {
+			const GDScriptParser::ConstantNode *cons = (const GDScriptParser::ConstantNode *)p_node;
+			sz += variant_to_string(cons->value);
+		} break;
+		case GDScriptParser::Node::TYPE_LOCAL_VAR: {
+			// const GDScriptParser::LocalVarNode *var = (const GDScriptParser::LocalVarNode *)p_node;
+			sz += draw_keyword("var ");
+			data.var_declaration = true;
+		} break;
+		case GDScriptParser::Node::TYPE_FUNCTION: {
+			const GDScriptParser::FunctionNode *func = (const GDScriptParser::FunctionNode *)p_node;
+			output_function(func, out);
+		} break;
+		case GDScriptParser::Node::TYPE_BUILT_IN_FUNCTION: {
+			const GDScriptParser::BuiltInFunctionNode *func = (const GDScriptParser::BuiltInFunctionNode *)p_node;
+			const char *func_name = GDScriptFunctions::get_func_name(func->function);
+			sz += func_name;
+		} break;
+		case GDScriptParser::Node::TYPE_SELF: {
+			sz += draw_keyword("self");
+		} break;
+		case GDScriptParser::Node::TYPE_ASSERT: {
+			sz += draw_keyword("assert ");
+			const GDScriptParser::AssertNode *ass = (const GDScriptParser::AssertNode *)p_node;
+			if (ass->condition) {
+				output_node(ass->condition, Output(out.indent, &sz), true);
+			}
+		} break;
+		case GDScriptParser::Node::TYPE_NEWLINE: {
+#if 1
+			sz += "\n";
+#else
+			sz += " ...\n";
+#endif
```

Comment:
I think the `#else` part (and the `#if`) can be removed here, unless it'll be toggled often at compile-time.